tiff: Make names more consistent with the spec. Note: API change.

mewmew opened the pull request:
Field names are derived from Section 2, Image File Directory (TIFF spec 6.0)
    Tag: "Id" seems appropriate even though it deviates from the spec, since
         name "Tag" is used to represent the struct.
    Type: Rename "Fmt" to "Type" to make this field consistent with the spec
    Count: Rename "Ncomp" to "Count" to make this field consistent with the
           The "output" struct in "tiff_test.go" already uses the name "coun
    Value/Offset: "Val" is consistent with the spec.

This pull request depends on/includes pull request 8 which fixes a bug in convertVals.

rwcarlsen commented:
Seems reasonable enough - especially since the API change is not likely to be part of the API that gets much use.

Commit:
tiff: Fix bug in convertVals, where floatVals was allocated more than once.

Commit:
tiff: Make names more consistent with the spec. Note: API change.

Field names are derived from Section 2, Image File Directory (TIFF spec 6.0)
    Tag: "Id" seems appropriate even though it deviates from the spec, since
         name "Tag" is used to represent the struct.
    Type: Rename "Fmt" to "Type" to make this field consistent with the spec
    Count: Rename "Ncomp" to "Count" to make this field consistent with the
           The "output" struct in "tiff_test.go" already uses the name "coun
    Value/Offset: "Val" is consistent with the spec.
